doc/demo/sparse_train/sparse_train.md

```diff
@@ -0,0 +1,291 @@
+# Sparse Training with PaddlePaddle
+
+This Document will guide you to understand the sparse design in PaddlePaddle.
+
+## Quick Start
+
+Assuming that you are familiar with PaddlePaddle. Here show train billion level parameters with sparse training. 
+
+### Prepare Data
+Given the Input data sample as follows:
+```
+0,0,7216737 1,18127654 1, ... 
+```
+The first slot ```0``` is just one flag, not one feature; the second slot ```0``` is  label; the third slot is feature id, ```7216737 1```means the ```7216737``` dimension is not zero, the ```1``` is just one flag, not one feature; the ```18127654 1``` is another non zero feature. The line data depicts one simple high dimension input data for classification task.
```

Comment:
the 'slot' here just is used to describe this sample, not related PaddlePaddle. I will add further emphasis on Dataprovider usage.